cnd/src/http_api/dial_addr.rs

```diff
@@ -20,5 +20,12 @@ pub async fn post_dial_addr(
     for addr in body.addresses {
         facade.dial_addr(addr).await;
     }
+
+    // Best effort, assume we got a connection.
```

Comment:
I don't understand, isn't the whole idea of this patch to _ensure_ we got a connection and not _assume_ it?

Reply:
We can't ensure we have a connection, the aim of this patch is to subscribe at a time when we can guess that we might have a connection. This patch tries to apply a bandaid solution to the problem that subscription before connection may be problematic, if it is not a good bandaid, or a fully thought out solution is deemed necessary, then we should totally do that. The issue that triggered this work is under discussion on github, I'm following it, that's why this is draft.

Thanks for asking though because I'm not convince that the gossipsub stuff is fully thought out in general - at least its not fully thought out by me.